src/test/java/seedu/address/logic/commands/DeleteStudentFromClassCommandTest.java

```diff
@@ -0,0 +1,154 @@
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.deletestudentfromclasscommands.DeleteStudentFromClassByEmailCommand;
+import seedu.address.logic.commands.deletestudentfromclasscommands.DeleteStudentFromClassByIdCommand;
+import seedu.address.logic.commands.deletestudentfromclasscommands.DeleteStudentFromClassByIndexCommand;
+import seedu.address.logic.messages.PersonMessages;
+import seedu.address.logic.messages.TutorialClassMessages;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.TutorialClass;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.StudentId;
+import seedu.address.testutil.PersonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteStudentFromClassCommand}.
+ */
+public class DeleteStudentFromClassCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private TutorialClass tutorialClass;
+
+    @BeforeEach
+    public void setUp() {
+        ModuleCode newModule = new ModuleCode(VALID_MODULE_AMY);
+        model.addModule(newModule);
+        TutorialClass newTutorialClass = new TutorialClass(VALID_TUTORIAL_AMY);
+        newModule.addTutorialClass(newTutorialClass);
+        tutorialClass = newTutorialClass;
+    }
+
+    @Test
+    public void execute_invalidStudent_fail() {
+        DeleteStudentFromClassByEmailCommand deleteStudentFromClassByEmailCommand =
+                new DeleteStudentFromClassByEmailCommand(new Email(INVALID_PERSON_EMAIL),
+                new ModuleCode(VALID_MODULE_AMY), new TutorialClass(VALID_TUTORIAL_AMY));
+        DeleteStudentFromClassByIdCommand deleteStudentFromClassByIdCommand =
+                new DeleteStudentFromClassByIdCommand(new StudentId(INVALID_PERSON_STUDENT_ID),
+                new ModuleCode(VALID_MODULE_AMY), new TutorialClass(VALID_TUTORIAL_AMY));
+        DeleteStudentFromClassByIndexCommand deleteStudentFromClassByIndexCommand =
+                new DeleteStudentFromClassByIndexCommand(Index.fromOneBased(1000),
+                new ModuleCode(VALID_MODULE_AMY), new TutorialClass(VALID_TUTORIAL_AMY));
+        assertCommandFailure(deleteStudentFromClassByEmailCommand, model,
+                String.format(PersonMessages.MESSAGE_PERSON_EMAIL_NOT_FOUND, INVALID_PERSON_EMAIL));
+        assertCommandFailure(deleteStudentFromClassByIdCommand, model,
+                String.format(PersonMessages.MESSAGE_PERSON_STUDENT_ID_NOT_FOUND, INVALID_PERSON_STUDENT_ID));
+        assertCommandFailure(deleteStudentFromClassByIndexCommand, model,
+                String.format(PersonMessages.MESSAGE_PERSON_INDEX_NOT_FOUND, 1000));
```

Comment:
I think the readability of this part can be improved. Perhaps a suggestion is that you can separate the different delete commands into blocks with linebreaks?